src/openvic-simulation/economy/FactoryProducer.cpp

```diff
@@ -37,3 +44,71 @@ fixed_point_t FactoryProducer::get_average_profitability_last_seven_days() const
 
 	return sum / (1 + profit_history_current);
 }
+
+void FactoryProducer::produce(
+	const fixed_point_t input_modifier, const fixed_point_t throughput_modifier, const fixed_point_t output_modifier
+) {
+	std::map<PopType const* const, fixed_point_t> employees_per_job {};
```

Comment:
Better to use `ordered_map` from `openvic-simulation/types/OrderedContainers.hpp`, we've switched to using it in the rest of the repo over the `std` maps and sets both for performance (it's a hash set) and ordering control (`std` maps only allow either no order or key comparison order).